src/service.js

```diff
@@ -0,0 +1,316 @@
+if(!global._babelPolyfill) { require('babel-polyfill'); }
+
+import Proto from 'uberproto';
+import mongoose from 'mongoose';
+import filter from 'feathers-query-filters';
+import errors from 'feathers-errors';
+import errorHandler from './error-handler';
+
+// Use native promises
+mongoose.Promise = global.Promise;
```

Comment:
I think being able to define your own promise implementation is a better choice and allows maximum choice for the developer to use what they need or want.

Not sure what versions of node this project is expecting to support over time, but working from the Node LTS guide here: 

https://github.com/nodejs/LTS/

If you'd like to support the maximum number of current LTS versions in the wild, you still need to support v.12 till Apr next year. 

If my assumptions up until now are correct, you need the option to pass in your own promise library to support v.12. 

I'm not claiming to be an expert though, so please double check my quick research.